core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -113,6 +129,36 @@ private[spark] class HistoryPage(parent: HistoryServer) extends WebUIPage("") {
       </div>
     UIUtils.basicSparkPage(content, "History Server")
   }
+  
+  private def getApplicationLevelList (appNattemptList: Iterable[ApplicationHistoryInfo])  ={
+    // Create HashMap as per the multiple attempts for one application. 
+    // If there is no attempt specific stuff, then
+    // do return false, to indicate the same, so that previous UI gets displayed.
+    var hasAttemptInfo = false
+    val appToAttemptInfo = new HashMap[String, ArrayBuffer[ApplicationHistoryInfo]]
+    for( appAttempt <- appNattemptList) {
```

Comment:
All throughout the code: `for (...)`, `if (...)`. Note the placement of the space.